test/common/router/route_fuzz_test.cc

```diff
@@ -86,7 +86,7 @@ DEFINE_PROTO_FUZZER(const test::common::router::RouteTestCase& input) {
     if (headers.ForwardedProto() == nullptr) {
       headers.insertForwardedProto().value(std::string("http"));
     }
-    auto route = config.route(headers, input.random_value());
+    auto route = config.route(headers, stream_info, input.random_value());
```

Comment:
CC @asraa this will be a place that we might want to start fuzzing as well; today it's effectively a boolean that routing depends upon (see the API protos above), but in the future it could get arbitrarily complicated.